addons/sales_team/models/crm_team.py

```diff
@@ -188,7 +188,7 @@ def _compute_member_warning(self):
                 team.member_warning = member_warning + " " + _("To add a Salesperson into multiple Teams, activate the Multi-Team option in settings.")
 
     def _search_member_ids(self, operator, value):
-        return [('crm_team_member_ids.user_id', operator, value)]
+        return [('crm_team_member_ids.user_id', operator, value), ('crm_team_member_ids.active', '=', True)]
```

Comment:
Have to say this is weird ... normally when using a model having an "active" flag, it is taken into account unless the specific context key 'active_test' is given. Is it because ORM optimization does joins and does not really respect the active flag @thiryr  ?

Reply:
I assumed it was because we were using `.user_id` yes
But I'll have a look to confirm

Reply:
@tde-banana-odoo 

Ok so I tracked this down. In c15c07c40517bfece5df66106772ed16eedcf6fa we prevent filtering of active id for so-called 'multi-path domains' like here when the comodel is m2o. So if you do `team_id.member_ids` here's what happens:

- team_id is m2o so we set `active_test` to False while we look up the subquery (i.e. the rest of the path)
- we look for member_ids on the team, which calls our custom search domain
- the custom search domain itself has a multi-path domain but it's m2m
- the multipath m2m is processed but the context of the original 'main' query is still active

Basically here
```
            elif operator in ('any', 'not any') and field.store and field.type == 'many2one':
                right_ids = comodel.with_context(active_test=False)._search(right)
                push((left, ANY_IN[operator], right_ids), model, alias)

            # Making search easier when there is a left operand as one2many or many2many
            elif operator in ('any', 'not any') and field.store and field.type in ('many2many', 'one2many'):
                right_ids = comodel.with_context(**field.context)._search(right)
                push((left, ANY_IN[operator], right_ids), model, alias)

```

The 'outer domain' (`team_id.member_ids`) goes to the first branch
Then the 'inner domain' (`crm_team_member_ids.user_id`) goes through the second one.

Except the context is never actually cleared between the two !

This means another possible fix is to set the attribute `context={'active_test': True}` explicitly on the `crm_team_member_ids` field. And indeed we see the test passes with that too.

This should be a fairly rare situation where we have 'multipath domains' multiple times in a row to resolve a single search query, just because multipath domains are rare. But it might warrant a real fix at some point...

Reply:
Thanks for the analysis by the way :D